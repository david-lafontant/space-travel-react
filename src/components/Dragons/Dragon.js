import 'bootstrap/dist/css/bootstrap.min.css';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Button from 'react-bootstrap/Button';
import Badge from 'react-bootstrap/Badge';
import { useDispatch } from 'react-redux';
import { PropTypes } from 'prop-types';
import { addDragonReservation, removeDragonReservation } from '../../redux/dragons/dragons';

const Dragon = (props) => {
  const dispatch = useDispatch();
  const {
    id, name, image, desc, reservation, type,
  } = props;
  // console.log(desc);
  const handleReserveClick = () => {
    dispatch(addDragonReservation(id));
  };

  const handleRemoveReserveClick = () => {
    dispatch(removeDragonReservation(id));
  };

  return (
    <Row id={id}>
      <Col sm={4} lg={4}>
        <img style={{ width: '100%', marginBottom: '50px' }} src={image} alt="Dragon" />
      </Col>
      <Col sm={8} lg={8}>
        <div style={{
          display: 'flex', gap: '6rem', marginBottom: '1.5rem', alignItems: 'center',
        }}
        >
          <h2>
            {name}
          </h2>
          <p style={{ display: 'flex', gap: '1rem', fontSize: '2rem' }}>
            <h3>Type:</h3>
            <h3>{type}</h3>
          </p>

        </div>

        <p style={{ fontSize: '1.5rem', marginBottom: '1.5rem' }}>
          {reservation && <Badge bg="warning" text="dark">Reserved</Badge>}
          {`${desc}`}
        </p>
        {reservation ? (
          <Button size="lg" variant="secondary" type="button" className="remove-reservation" onClick={handleRemoveReserveClick}>
            Cancel Reservation
          </Button>
        ) : (
          <Button size="lg" variant="primary" type="button" className="add-reservation" onClick={handleReserveClick}>
            Reserve Dragon
          </Button>
        )}
      </Col>
    </Row>
  );
};

Dragon.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  desc: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  reservation: PropTypes.bool.isRequired,
};

export default Dragon;
