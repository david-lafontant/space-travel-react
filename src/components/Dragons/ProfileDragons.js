import { useDispatch } from 'react-redux';
import { PropTypes } from 'prop-types';
import Button from 'react-bootstrap/Button';
import { removeDragonReservation } from '../../redux/dragons/dragons';

const ProfileDragons = (props) => {
  const dispatch = useDispatch();
  const { id, name } = props;

  const handleRemoveReservation = () => {
    dispatch(removeDragonReservation(id));
  };

  return (
    <tr>
      <th>{name}</th>
      <td><Button type="button" variant="danger" className="remove-reservation" onClick={handleRemoveReservation}>Cancel Reservation</Button></td>
    </tr>
  );
};

ProfileDragons.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};

export default ProfileDragons;
