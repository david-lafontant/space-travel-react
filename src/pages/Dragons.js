import { useSelector } from 'react-redux';
import Dragon from './Dragon';

const Rockets = () => {
  const dragonList = useSelector((state) => state.dragonsReducer);
  return (
    <ul>
      {dragonList.map((item) => (
        <Dragon
          desc={item.desc}
          key={item.id}
          id={item.id}
          type={item.type}
          name={item.name}
          image={item.flickr_images}
          reservation={item.reservation}
        />
      ))}
    </ul>
  );
};

export default Rockets;
