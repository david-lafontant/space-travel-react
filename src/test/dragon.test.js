import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import elementStore from '../redux/configureStore';
import Dragons from '../pages/Dragons';

// ARRANGE
const dragons = [
  {
    desc: 'The best capsule in the solar system',
    key: 'besto',
    id: 'besto',
    type: 'capsule',
    name: 'frendo',
    image: 'url',
    reservation: false
  },
];

describe('Test render the Dragon Component', () => {
  it('Dragons section renders a hardcoded object, passed as a prop', () => {
    // ACT
    const { container } = render(
      <Provider store={elementStore}>
        <Dragons key={dragons[0].id} dragon={dragons[0]} />
      </Provider>,
    );

    // ASSERT
    expect(container).toMatchSnapshot();
  });
});
