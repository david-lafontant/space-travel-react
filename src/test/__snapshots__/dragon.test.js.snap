// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render the Dragon Component Dragons section renders a hardcoded object, passed as a prop 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-lg-4 col-sm-4"
    >
      <img
        alt="Dragon"
        style="width: 100%; margin-bottom: 50px;"
      />
    </div>
    <div
      class="col-lg-8 col-sm-8"
    >
      <div
        style="display: flex; gap: 6rem;"
      >
        <h2 />
        <p>
          Type:
        </p>
      </div>
      <p>
        undefined
      </p>
      <button
        class="add-reservation btn btn-primary"
        type="button"
      >
        Reserve Dragon
      </button>
    </div>
  </div>
</div>
`;
